test/gleam/erlang/file_test.gleam

```diff
@@ -0,0 +1,44 @@
+import gleam/string
+import gleam/erlang/file
+
+pub fn successful_read_test() {
+  assert Ok(<<"Hello, World!\n":utf8>>) =
+    file.read_file("test/fixtures/success.txt")
+}
+
+pub fn unsuccessful_read_test() {
+  assert Error(file.Enoent) = file.read_file("test/fixtures/does_not_exist.txt")
+}
+
+pub fn successful_write_test() {
+  let path = tmp_path("write_test.txt")
+
+  assert Ok(Nil) = file.write_file(path, <<"Hello, World!":utf8>>)
+  assert Ok(<<"Hello, World!":utf8>>) = file.read_file(path)
+
+  // Cleanup
+  file.delete_file(path)
+}
+
+pub fn unsuccessful_write_test() {
+  let path = tmp_path("doesnt_exist/write_test.txt")
+
+  assert Error(file.Enoent) = file.write_file(path, <<"Hello, World!":utf8>>)
+}
+
+pub fn successful_delete_test() {
+  let path = tmp_path("example.txt")
+  file.write_file(path, <<>>)
+
+  assert Ok(Nil) = file.delete_file(path)
+}
+
+pub fn unsuccessful_delete_test() {
+  let path = tmp_path("missing.txt")
+
+  assert Error(file.Enoent) = file.delete_file(path)
+}
+
+fn tmp_path(filename: String) {
+  string.concat(["test/tmp", filename])
```

Comment:
Could we gitignore this?

Reply:
The whole task of testing file system operations was a bit contrived. In order to test, say, `delete`, the file either needs to exist before the test runs and be rewritten at the end of the test or else to be written at the beginning of the test and later deleted. For the `write_bits` test, the file's contents must be verified by calling `read_bits` and then the file must be deleted during test teardown. Any one test that ostensibly tests a single function from the file module actually winds up invoking several, meaning that any error in a single function will likely cause all tests to fail.

The `read_bits` test has a similar problem&mdash;ish. If the file already existed, then it could be read without needing to invoke `write_bits` first, which is why the `test/tmp` directory is not ignored and has a single text file in it.

That said, it seems like the testing strategy could be reconsidered. A few ideas are:
- Embrace the dependencies, write several tests that utilize `read_bits`, `write_bits`, and `delete` without claiming that any test belongs to just a single function.
- Mock Erlang's `file` module with `meck` or similar; e.g. mock out `file:read_file` to always return `{ok, "Hello World!"}`. Our tests would be more "unit test" than "integration test", and we assume that our mock faithfully replicates Erlang's `file`. `meck` specifically comes with the downside that any module patching it does applies globally, which means tests using it aren't async-safe in Elixir land.
- Mox-style mocking where we substitute in an alternate module to be used for our tests other than `file`. This solution is very popular in Elixir land, but requires a good lot of work to set up, and I've never attempted something like this in Erlang/Gleam.

Reply:
I think it's OK to embrace the dependencies. It may be more confusing to debug a failure, but these functions are quite simple so I think it won't be an issue.

Reply:
Ah, to cycle back to the original question: I'm keeping the empty (well, with a `.gitkeep` file in it) `test/tmp` directory, as this is where these test will write files. I will also gitignore the directory once it's been added to git such that any trash that winds up in there won't be accidentally committed. Theoretically these could be written anywhere, including `/tmp`, but I figured this makes the most sense.